### ISSUE TITLE
Show total IO in systemctl

### ISSUE THREAD

**Issue by ghost** [systemd/systemd#9768]
**systemd version the issue has been seen with**
 239

**Used distribution**

Debian unstable in systemd-nspawn on Ubuntu 18.04 LTS

**Expected behaviour you didn't see**

With `IOAccounting=yes` and existence of `io.stat` in a directory corresponding to a unit,
`systemctl status someunit.service` etc. should show the information stored in `io.stat`

**Unexpected behaviour you saw**

Systemd nor systemctl reads information from `io.stat`. I run

```
fgrep -n "io.stat" `find -type f -print`
```

on the latest source of systemd, and found that `io.stat` is only referred in `src/cgtop/cgtop.c:334`,
so there is **absolutely no hope** for `systemctl` to show information in `io.stat`.

In addition no option to `systemd-cgtop` lets `systemd-cgtop` to print information from `io.stat`.
Then, what is the intention of `IOAccouting=yes` in the unified cgroup hierarchy??
I cannot understand what is meant by systemd.resource-control manpage.
What is the expected behavior **?????????????????????????**

I did the following:

```
root@gtx1080ti-unstable:/etc/systemd/system# systemctl show user-0.slice | fgrep Accounting
CPUAccounting=yes
IOAccounting=yes
BlockIOAccounting=no
MemoryAccounting=yes
TasksAccounting=yes
IPAccounting=yes

root@gtx1080ti-unstable:/etc/systemd/system# cat /sys/fs/cgroup/user.slice/user-0.slice/io.stat 
8:0 rbytes=1425408 wbytes=20480 rios=36 wios=1

root@gtx1080ti-unstable:/etc/systemd/system# systemctl status -l  user-0.slice
● user-0.slice - User Slice of UID 0
   Loaded: loaded
  Drop-In: /lib/systemd/system/user-.slice.d
           └─10-defaults.conf
           /etc/systemd/system/user-.slice.d
           └─accounting.conf
   Active: active since Wed 2018-08-01 16:48:52 JST; 23min ago
    Tasks: 6 (limit: 10813)
   Memory: 11.5M
      CPU: 1.578s
   CGroup: /user.slice/user-0.slice
           ├─session-345.scope
           │ ├─1866 sshd: root@pts/0
           │ ├─1879 -bash
           │ ├─1973 systemctl status -l user-0.slice
           │ └─1974 pager
           └─user@0.service
             └─init.scope
               ├─1869 /lib/systemd/systemd --user
               └─1870 (sd-pam)
```

**Steps to reproduce the problem**

See above.

**Comment by keszybz** [comment]
Showing the statistics in systemctl or cgtop is nice, but that's just a secondary effect. The primary effect of those settings is to configure the cgroup attributes. IOAccounting= can be used to turn on io controller without configuring weights. There might be bugs in this, so if you see that the setting does not have the expected effect on cgroup attributes, make a bug report showing what is wrong.

A second issue is that systemctl and/or cgtop is not showing some stuff it could. Here cgtop shows "Input/s Output/s" and it seems to have some useful values. But showing io stats for systemctl makes less sense than for cgtop. Those stats are very transient, so showing them "live" as cgtop does is useful, but showing them in a static systemctl output less so.

**Comment by ryutaroh-matsumoto** [comment]
@keszybz Thank you for your response. As I was requested reporter-feedback, I report the below.

>  But showing io stats for systemctl makes less sense than for cgtop. Those stats are very transient, so showing them "live" as cgtop does is useful, but showing them in a static systemctl output less so.

I think the above interpretation is logically inconsistent with what is done with `CPUAccounting=yes`. When `CPUAccounting=yes`, `systemctl status` shows the amount of CPU time from `cpu.stat`, which seems `transient` state. Why CPU usage time is shown by `systemctl status` while `io.stat` is not shown by `systemctl status`? It seems logically inconsistent to me.

> There might be bugs in this, so if you see that the setting does not have the expected effect on cgroup attributes, make a bug report showing what is wrong.

WIth regard to this issue 9768, `systemd-cgtop` does not show "Input/s Output/s" even when `IOAccounting=yes` and there exists `io.stat` in the corresponding cgroup. I am aware that IO controller is enabled by `IOAccounting=yes`, but there seems no way for a user to read out `io.stat` by some systemd tools (of course we can `cat io.stat`). Do you expect an ordinary user to `cat io.stat`? If it is, that's fine and there is no problem.

Second, with regard to [9673](https://github.com/systemd/systemd/issues/9763), IO controller is not enabled even if `IOAccounting=yes` is specified, under some condition. It seems a separate issue and it was filed separately as [9673](https://github.com/systemd/systemd/issues/9763).

I hope the above is sufficient to remove `needs-reporter-feedback` tag.

**Comment by poettering** [comment]
The fact that systemctl status doesn't show IO accounting information is that noone sat down so far to add it. Would be happy to merge a patch for that. We currently show CPU, Memory, IP, Tasks accounting, only IO is missing.

**Comment by ryutaroh-matsumoto** [comment]
@keszybz 
> It shows here. If it doesn't work for you, then there's a bug.

Umm... Thank you, but while `dd if=/dev/zero of=/dev/null`, I run `systemd-cgtop` and looked `user-0.slice`. `%CPU` is roughly 100% but `Input/s` and `Output/s` are both `-`. `IOAccounting=yes` is not supported for `user-0.slice` in `systemd-cgtop`? The source code of cgtop seems to intend reading information from `io.stat`...

**Comment by ryutaroh-matsumoto** [comment]
I submitted PR #9798 to meet this RFE.

**Comment by poettering** [comment]
Was fixed in #12074.
